netflix-roulette/src/pages/movieDetalis/MovieDetailsContainer.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { Formik } from "formik";
+import {
+  AddOrEditMovie,
+  AddOrEditMoviePayload,
+  MoviesState,
+  SetIsAddMovieDialogVisible,
+  SetIsAddMovieDialogVisiblePayload,
+  SetMovieToEditOrView,
+  SetMovieToEditOrViewPayload,
+} from "@store/interfaces";
+import { MovieDetails } from "./MovieDetails";
+import connectedToStore from "./connectedToStore";
+import { formMovieData, validate } from "./utils";
+
+interface MovieDetailsProps extends Partial<MoviesState> {
+  setIsAddMovieDialogVisible: (
+    payload: SetIsAddMovieDialogVisiblePayload
+  ) => SetIsAddMovieDialogVisible;
+  setMovieToEdit: (
+    payload: SetMovieToEditOrViewPayload
+  ) => SetMovieToEditOrView;
+  addMovie: (payload: AddOrEditMoviePayload) => AddOrEditMovie;
+  editMovie: (payload: AddOrEditMoviePayload) => AddOrEditMovie;
+}
+
+const MovieDetailsComponent = ({
+  setIsAddMovieDialogVisible,
+  isAddMovieDialogVisible,
+  movieToEdit,
+  setMovieToEdit,
+  addMovie,
+  editMovie,
+}: MovieDetailsProps) => {
+  const handleDialogClose = React.useCallback(() => {
+    isAddMovieDialogVisible &&
+      setIsAddMovieDialogVisible({ isAddMovieDialogVisible: false });
+    movieToEdit && setMovieToEdit({ movie: null });
+  }, [isAddMovieDialogVisible, movieToEdit]);
+
+  const onSubmit = React.useCallback(
```

Comment:
Not sure, whether this can be an issue here, however, worth mentioning that using `useCallback()` for memoizing edited data from API, if used where not appropriate, may lead to inconsistencies between front-end and back-end data.